contracts/ArrowToken.sol

```diff
@@ -1,29 +1,42 @@
-pragma solidity ^0.8.11;
-
 // SPDX-License-Identifier: MIT
+pragma solidity ^0.8.11;
 
-import "@openzeppelin/contracts/access/Ownable.sol";
-import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC20/ERC20Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
 
 /**
-    @title Arrow Token implementation
- */
-contract ArrowToken is ERC20, Ownable {
+    @title Arrow Token implementation that is upgradeable
+    @notice Create ARROW tokens of an `initialSupply` amount
+    @dev If we were to upgrade the initializer by introducing new variables, we need to create a normal function like `initializeV2` (https://forum.openzeppelin.com/t/uups-proxies-tutorial-solidity-javascript/7786/53), and make sure to include all existing variables from the current version of the contract to avoid storage collision
+*/
+contract ArrowToken is
+    Initializable,
+    ERC20Upgradeable,
+    OwnableUpgradeable,
+    UUPSUpgradeable
+{
+    // implement the UUPS interface
+    function _authorizeUpgrade(address) internal override onlyOwner {}
 
-    /// @notice This function assigns msg.sender as the token admin and mints them initial supply
-    constructor(uint256 initialSupply, string memory _name, string memory _symbol)
-        ERC20(_name, _symbol)
-    {
-        _mint(msg.sender, initialSupply);
-    }
+    /** 
+        @notice Mint an `_initialSupply` amount of tokens upon contract creation
+    */
+    function initialize(uint256 _initialSupply) public initializer {
+        __ERC20_init("Arrow", "ARROW");
+        __Ownable_init();
+        __UUPSUpgradeable_init();
 
-    /**
-        Mints new tokens.
+        _mint(msg.sender, _initialSupply);
+    }
 
-        Only accepts calls from the contract owner.
-     */
-    function mint(address to, uint256 amount) public onlyOwner
-    {
-        _mint(to, amount);
+    /**     
+        @notice Mint and issue new tokens to a specified address
+        @param _to the address to issue new tokens to
+        @param _amount the amount of tokens to be mint and transfered to `_to` 
+   */
+    function mint(address _to, uint256 _amount) public onlyOwner {
+        _mint(_to, _amount);
```

Comment:
Let's do the minimum viable change. There's no need to alter the mint function in this PR. We can add documentation in a separate PR.